2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,382 @@
-        assertTrue(integerQueue.isEmpty());
+    void pollFromEmptyQueue() {
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = this.integerQueue.poll();
+
+        assertThat(isEmpty).isEqualTo(true);
+        assertThat(firstElement).isEqualTo(null);
     }
 
     @Test
     @Order(9)
-    void isEmpty() {
-        integerQueue.add(3);
-        integerQueue.add(9);
+    void pollFromNonEmptyQueue() {
+        addIntElementToQueue(11);
+        addIntElementToQueue(111);
+
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = this.integerQueue.poll();
+
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(11);
+    }
+
+    @Test
+    @Order(10)
+    void pollLastElement() {
+        addIntElementToQueue(12);
+        addIntElementToQueue(10);
+
+        int sizeBeforePollOperation = getInternalSize();
+        boolean isEmptyBeforePoll = isEmptyQueue();
+        Integer firstElement = this.integerQueue.poll();
+        Integer secondElement = this.integerQueue.poll();
+        boolean isEmptyAfterPoll = isEmptyQueue();
+        int sizeAfterPollOperation = getInternalSize();
+
+        assertThat(isEmptyBeforePoll).isEqualTo(false);
+        assertThat(isEmptyAfterPoll).isEqualTo(true);
+
+        assertThat(sizeBeforePollOperation).isEqualTo(2);
+        assertThat(sizeAfterPollOperation).isEqualTo(0);
+
+        assertThat(firstElement).isEqualTo(12);
+        assertThat(secondElement).isEqualTo(10);
+    }
+
+
+    @Test
+    @Order(11)
+    void addElementIntoEmptyQueue() {
+        int sizeBeforeAdd = getInternalSize();
+        boolean isEmptyBeforeAdd = isEmptyQueue();
+
+        integerQueue.add(228);
+        int sizeAfterAdd = getInternalSize();
+        boolean isEmptyAfterAdd = isEmptyQueue();
+        Integer elementValue = (Integer) pollElementFromQueue();
+
+        assertThat(sizeBeforeAdd).isEqualTo(0);
+        assertThat(sizeAfterAdd).isEqualTo(1);
+
+        assertThat(isEmptyBeforeAdd).isEqualTo(true);
+        assertThat(isEmptyAfterAdd).isEqualTo(false);
+
+        assertThat(elementValue).isEqualTo(228);
+    }
 
-        assertFalse(integerQueue.isEmpty());
+    @Test
+    @Order(12)
+    void addElementIntoNotEmptyQueue() {
+        addIntElementToQueue(12);
+
+        int sizeBeforeAdd = getInternalSize();
+        boolean isEmptyBeforeAdd = isEmptyQueue();
+
+        integerQueue.add(111);
+        int sizeAfterAdd = getInternalSize();
+        boolean isEmptyAfterAdd = isEmptyQueue();
+        Integer firstValue = (Integer) pollElementFromQueue();
+        Integer secondValue = (Integer) pollElementFromQueue();
+
+        assertThat(sizeBeforeAdd).isEqualTo(1);
+        assertThat(sizeAfterAdd).isEqualTo(2);
+
+        assertThat(isEmptyBeforeAdd).isEqualTo(false);
+        assertThat(isEmptyAfterAdd).isEqualTo(false);
+
+        assertThat(firstValue).isEqualTo(12);
+        assertThat(secondValue).isEqualTo(111);
+    }
+
```

Comment:
I suggest adding the last test-method that not use any features from `Reflection Api`.

Reply:
Can you explain more about your idea? Our essential approach is using only one method of object in an appropriate test.